ipfix/node.c

```diff
@@ -327,6 +502,130 @@ ipfix_node_fn (vlib_main_t * vm,
   return frame->n_vectors;
 }
 
+static void ipfix_free_v10_packet(netflow_v10_data_packet_t *packet)
+{
+  netflow_v10_data_set_t *set;
+  vec_foreach(set, packet->sets) {
+    free(set->data);
+  };
+  vec_free(packet->sets);
+}
+
+static void ipfix_build_v10_packet(ipfix_ip4_flow_value_t *record,
+                                   netflow_v10_data_packet_t *packet)
+{
+  netflow_v10_template_t template;
+  ipfix_make_v10_template(&template);
+
+  clib_warning("%U", format_netflow_v10_template, &template);
+
+  struct timespec current_time_clock;
+  clock_gettime(CLOCK_REALTIME, &current_time_clock);
+
+
+  packet->sets = 0;
+  packet->header.version = ntohs(10);
+  packet->header.timestamp = ntohs(current_time_clock.tv_sec);
+
+  netflow_v10_template_set_t *set;
+  netflow_v10_field_specifier_t *field;
+  vec_foreach(set, template.sets) {
+    u64 data_size = 0;
+    vec_foreach(field, set->fields) {
+      data_size = data_size + field->size;
+    }
+
+    netflow_v10_data_set_t active_set;
+    active_set.data = malloc(data_size);
+    void *ptr = (size_t)active_set.data;
+
+    vec_foreach(field, set->fields) {
```

Comment:
I wonder if it would make sense to store the mapping of template fields to portions of the flow record by storing a mapping of fields to offsets in the flow record. That way this code would only deal with one case, but there would be an extra data structure with that mapping. WDYT?

Reply:
I just want to verify I've understood correctly.

We'd add an additional field to `netflow_v10_field_specifier_t` which specifies the offset of `ipfix_ip4_flow_value_t` struct where the field lies so you can just do, for example:

```c
memcpy(ptr, (void *)((u8 *)record + field->record_offset), field->size);
```

I think that'd work well and it'd certainly improve the code if we ended up supporting even more attributes.